File: ゼミ用/05製造/api/src/graph/request/SearchActorRequest.php
<?php

/**
 * 時系列検索処理の値を保持する
 */
class SearchActorRequest {
    private $actorId;
    private $actorName;
    private $actorInfo;
    private $opusId;
    private $timeId;
    private $groupId;
    private $userId;
    private $offset = 0;
    private $limit = 100;
    private $errorMsg = array();

    /**
     * 時系列検索処理の値を格納します
     */
    public function __construct() {
    }

    public function getActorId() {
        return $this->actorId;
    }

    public function setActorId($actorId) {
        $this->actorId = $actorId;
    }

    public function getActorName() {
        return $this->actorName;
    }

    public function setActorName($actorName) {
        $this->actorName = $actorName;
    }

    public function getActorInfo() {
        return $this->actorInfo;
    }

    public function setActorInfo($actorInfo) {
        $this->actorInfo = $actorInfo;
    }

    public function getOpusId() {
        return $this->opusId;
    }

    public function setOpusId($opusId) {
        $this->opusId = $opusId;
    }

    public function getTimeId() {
        return $this->timeId;
    }

    public function setTimeId($timeId) {
        $this->timeId = $timeId;
    }

    public function getGroupId() {
        return $this->groupId;
    }

    public function setGroupId($groupId) {
        $this->groupId = $groupId;
    }

    public function getUserId() {
        return $this->userId;
    }

    public function setUserId($userId) {
        $this->userId = $userId;
    }

    public function getOffset() {
        return $this->offset;
    }

    public function setOffset($offset) {
        $this->offset = $offset;
    }

    public function getLimit() {
        return $this->limit;
    }

    public function setLimit($limit) {
        $this->limit = $limit;
    }

    public function getErrorMsg() {
        return $this->errorMsg;
    }

    /**
     * バリデーションチェックをします
     */
    public function validation() {
        $validationFlg = false;

        try {

            // ユーザIDの文字数チェック
            if (strlen($this->userId) != Constant::USER_ID_DIGIT + strlen(Constant::USER_ID_STR)) {
                array_push($this->errorMsg, 'エラー ：ユーザIDは' . Constant::USER_ID_DIGIT + strlen(Constant::USER_ID_STR) . '文字で入力してください');
                $validationFlg = true;
            }
        } catch (Exception $e) {
            $validationFlg = true;
        }

        return $validationFlg;
    }
}

File: ゼミ用/05製造/eeldev/eelPy/api/request/SearchActorRequest.py
from ..common.Constant import Constant


# /**
# * 登場人物検索処理の値を保持する
# */
class SearchActorRequest:
    errorMsg = []

    # /**
    # * 登場人物検索処理の値を格納します
    # */
    def __init__(self, actorId, actorName, actorInfo, actorImg, opusId, timeId, groupId, userId, offset, limit, version):
        self.actorId = actorId
        self.actorName = actorName
        self.actorInfo = actorInfo
        self.actorImg = actorImg
        self.opusId = opusId
        self.timeId = timeId
        self.groupId = groupId
        self.userId = userId
        self.offset = offset
        self.limit = limit
        self.version = version

    def getActorId(self):
        return self.actorId

    def getActorName(self):
        return self.actorName

    def getActorInfo(self):
        return self.actorInfo

    def getActorImg(self):
        return self.actorImg

    def getOpusId(self):
        return self.opusId

    def getTimeId(self):
        return self.timeId

    def getGroupId(self):
        return self.groupId

    def getUserId(self):
        return self.userId

    def getOffset(self):
        return self.offset

    def getLimit(self):
        return self.limit

    def getErrorMsg(self):
        return self.errorMsg

    # /**
    # * バリデーションチェックをします
    # */
    def validation(self):
        validationFlg = False

        try:
            # ユーザIDの文字数チェック
            if (Constant.USER_ID_DIGIT + len(Constant.USER_ID_STR) < len(self.userId)):
                self.errorMsg.append(
                    'エラー ：ユーザIDは' + Constant.USER_ID_DIGIT + len(Constant.USER_ID_STR) + '文字以内で入力してください')
                validationFlg = True

        except Exception as e:
            validationFlg = True

        return validationFlg
